othello/moderator/moderator.py

```diff
@@ -0,0 +1,114 @@
+import enum
+
+from . import constants, utils
+
+
+class Player(enum.Enum):
+    BLACK = "x"
+    WHITE = "o"
+
+
+PLAYERS = {
+    0: Player.WHITE,
+    1: Player.BLACK,
+}
+
+INITIAL_BOARD = "...........................ox......xo..........................."
+
+
+class InvalidMoveError(RuntimeError):
+    def __init__(self, board, player, move):
+        self.code = utils.UserError.INVALID_MOVE.value[0]
+        self.message = utils.UserError.INVALID_MOVE.value[1].format(
+            move=move, player=PLAYERS[player].value, board=utils.binary_to_string(board)
+        )
+
+    def __str__(self):
+        return self.message
+
+
+class Moderator:
+    def __init__(self):
+        self.board = constants.INITIAL
+        self.game_over = False
+        self.current_player = constants.BLACK
+
+    def is_game_over(self):
+        return self.game_over
+
+    def get_board(self):
+        return utils.binary_to_string(self.board)
+
+    def outcome(self):
+        if self.is_game_over():
+            score = self.score()
+            return (
+                PLAYERS[constants.BLACK].value
+                if score > 0
+                else PLAYERS[constants.WHITE].value
+                if score < 0
+                else "T"
+            )
+        return False
+
+    def possible_moves(self):
```

Comment:
Please heavily comment this method.

Reply:
Additionally, is there any reason it couldn't accept the player to find moves for as an optional parameter (defaulting to `self.current_player` if `None` is given? That would simplify `check_game_over()`.